### PR TITLE
Return validation results from verify_compact

### PR DESCRIPTION
**Pull request by jschlyter**

### CODE REVIEW

File: src/jwkest/jws.py [hunk 1]
```diff
@@ -566,7 +578,7 @@ def verify_compact(self, jws, keys=None, allow_none=False, sigalg=None):
                     logger.debug(
                         "Verified message using key with kid=%s" % key.kid)
                     self.msg = jwt.payload()
-                    return self.msg
+                    return (self.msg, {'key': key})
```
Comment (jschlyter) [inline src/jwkest/jws.py]: One could also consider returning only a dictionary with both msg and key.

**Comment by rohe** [comment]
I think that would be the way to go.

> 7 okt. 2016 kl. 16:31 skrev Jakob Schlyter <notifications@github.com>:
> 
> @jschlyter commented on this pull request.
> 
> In src/jwkest/jws.py:
> 
> > @@ -566,7 +578,7 @@ def verify_compact(self, jws, keys=None, allow_none=False, sigalg=None):
>                      logger.debug(
>                          "Verified message using key with kid=%s" % key.kid)
>                      self.msg = jwt.payload()
> -                    return self.msg
> +                    return (self.msg, {'key': key})
> 
> One could also consider returning only a dictionary with both msg and key.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

-- Roland
"Education is the path from cocky ignorance to miserable uncertainty.” - Mark Twain

**Comment by jschlyter** [comment]
We now return a dict.

**Comment by rohe** [comment]
Have you looked at the tests ?

**Comment by jschlyter** [comment]
I have now. Also changed to that JWS self.key contains the last key.

**Comment by rohe** [comment]
So, if self.key contains the key that was used to verify the signature is there still
a need for getting it return ?
Having the method return a dict changes the API in a not backward compatible way so
if it can be avoided I’d prefer that.

> 8 okt. 2016 kl. 19:13 skrev Jakob Schlyter <notifications@github.com>:
> 
> I have now. Also changed to that JWS self.key contains the last key.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

-- Roland
"Education is the path from cocky ignorance to miserable uncertainty.” - Mark Twain

**Comment by jschlyter** [comment]
I'm good with only setting self.key. Want a PR with only that?

--
Sent from my iPhone, hence this mail might be briefer than normal.

**Comment by jschlyter** [comment]
Although do note that the API was not changed, the new verify_compact_verbose() returns the dict and the old verify_compact() still returns the msg only.

But, isn't storing results in the object itself might be dangerous for concurrency? If two validations happens simultanetously with the same JWS object, they will overwrite each others results.

**Comment by rohe** [comment]
Ah, right ! Sorry about that !

> 10 okt. 2016 kl. 20:33 skrev Jakob Schlyter <notifications@github.com>:
> 
> Although do note that the API was not changed, the new verify_compact_verbose() returns the dict and the old verify_compact() still returns the msg only.
> 
> But, isn't storing results in the object itself might be dangerous for concurrency? If two validations happens simultanetously with the same JWS object, they will overwrite each others results.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

-- Roland
"Education is the path from cocky ignorance to miserable uncertainty.” - Mark Twain

**Pull request merged**